EKF/estimator_interface.cpp

```diff
@@ -125,16 +138,17 @@ bool EstimatorInterface::checkIfVehicleAtRest(float dt, const imuSample &imu)
 
 void EstimatorInterface::setMagData(const magSample &mag_sample)
 {
-	if (!_initialised || _mag_buffer_fail) {
+	if (!_initialised) {
 		return;
 	}
 
 	// Allocate the required buffer size if not previously done
-	// Do not retry if allocation has failed previously
-	if (_mag_buffer.get_length() < _obs_buffer_length) {
-		_mag_buffer_fail = !_mag_buffer.allocate(_obs_buffer_length);
+	if (_mag_buffer == nullptr && !_control_status.flags.in_air) {
```

Comment:
So we do retry to allocate if is previously failed? Why didn't we do that before?

Reply:
Spamming the error and potential performance issues if you get stuck trying (and failing) to allocate in flight. This allows it to re-try as needed up until flight. I'm also going to add a flag to rate limit the allocation failure print message.

Note this really isn't a common case we need to worry about  (especially past stm32f4).